sql/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java

```diff
@@ -237,4 +245,42 @@ public void testNonDeterministicFunctionsAlwaysNew() throws Exception {
         assertThat(fn1, is(equalTo(fn2)));
         assertThat(fn1, is(not(equalTo(fn3))));
     }
+
+    @Test
+    public void testInPredicateWithSubqueryIsRewrittenToAnyEq() {
+        TestingDocTableInfoFactory docTableInfoFactory = new TestingDocTableInfoFactory(
+            ImmutableMap.of(
+                T3.T1_INFO.ident(), T3.T1_INFO,
+                T3.T2_INFO.ident(), T3.T2_INFO
+            )
+        );
+        Map<QualifiedName, AnalyzedRelation> sources = ImmutableMap.of(
+            new QualifiedName("t1"), T3.TR_1,
+            new QualifiedName("t2"), T3.TR_2
+        );
+        StatementAnalysisContext statementAnalysisContext = new StatementAnalysisContext(
+            SessionContext.SYSTEM_SESSION,
+            paramTypeHints,
+            Operation.READ,
+            null
+        );
+        Schemas schemas = new Schemas(
+            Settings.EMPTY,
+            ImmutableMap.of(
+                Schemas.DEFAULT_SCHEMA_NAME,
+                new DocSchemaInfo(Schemas.DEFAULT_SCHEMA_NAME, clusterService, functions, null, docTableInfoFactory)),
+            clusterService,
+            new DocSchemaInfoFactory(docTableInfoFactory, functions, null)
+        );
+        RelationAnalyzer analyzer = new RelationAnalyzer(clusterService, functions, schemas);
+        ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
```

Comment:
@mfussenegger unfortunately I see no other way than this to test it since the analyzer requires a `SubqueryAnalyzer` https://github.com/crate/crate/blob/2.1/sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java#L760 otherwise it won't be able to process the tree.